pricegraph/src/orderbook/scalar.rs

```diff
@@ -0,0 +1,159 @@
+//! This module contains definitions for measurement scalars used by the
+//! orderbook graph representation.
+
+use crate::encoding::PriceFraction;
+use crate::FEE_FACTOR;
+use std::cmp;
+use std::ops;
+
+/// An exchange price. Limit prices on the exchange are represented by a
+/// fraction of two `u128`s representing a buy and sell amount. These limit
+/// prices implicitly include fees, that is they must be respected **after**
+/// fees are applied. As such, the actual exchange rate for a trade (the ratio
+/// of executed buy amount over executed sell amount) is strictly greater than
+/// and **never equal to** the limit price of an order.
+///
+/// Prices are guaranteed to be a strictly positive finite real numbers.
+#[derive(Clone, Copy, Debug, PartialEq, PartialOrd)]
+pub struct Price(f64);
+
+impl Price {
+    /// Creates a new price from a `f64` value. Returns `None` if the price
+    /// value is not valid. Specifically, it must be in the range `(0, +∞)`.
+    pub fn new(value: f64) -> Option<Self> {
```

Comment:
Any reason to not use try_from and into traits here?

Reply:
Not really, just was lazy to implement an error type for it. Rust standard has some wrapper number types for non-zero numbers that also follow this `new(inner) -> Option<Self>` pattern so I just went with it (https://doc.rust-lang.org/std/num/struct.NonZeroU64.html#method.new). If you think there is value in adding an error type for invalid prices and implementing `TryFrom`, we can definitely do that (and I think it would be very correct).